### PR TITLE
Modify openQA install to work on SLE

### PR DESCRIPTION
**Pull request by baierjan**
- Related ticket: https://progress.opensuse.org/issues/127286
- Needles: TBD
- Verification run:
    - http://dagr.qam.suse.cz/tests/83
    - http://dagr.qam.suse.cz/tests/85

### CODE REVIEW

**Review by okurz — CHANGES_REQUESTED** [review]
Except for

```
        add_suseconnect_product('PackageHub') if is_sle;
```

it looks like submitting a new version of openQA including openqa-bootstrap to PackageHub should suffice to not need the custom setup, isn't it?

File: tests/openqa/install/openqa_bootstrap.pm [hunk 1]
```diff
@@ -20,8 +22,65 @@ sub run {
...
+            # Download latest install scripts
+            assert_script_run('cd /tmp');
+            assert_script_run('wget https://raw.githubusercontent.com/os-autoinst/openQA/master/script/openqa-bootstrap');
+            assert_script_run('wget https://raw.githubusercontent.com/os-autoinst/openQA/master/script/fetchneedles');
+            assert_script_run('chmod +x openqa-bootstrap fetchneedles');
+
+            # Patch them (will be upstreamed)
+            zypper_call('in patch');
+            assert_script_run('wget ' . autoinst_url('/data/openqa-bootstrap.patch'));
+            assert_script_run('patch < openqa-bootstrap.patch');
```
[the remark this answers is not in the document]
Comment (baierjan) [reply tests/openqa/install/openqa_bootstrap.pm:39]: Ah, right.. missing file. It fixes other things, but nothing we cannot handle in our upstream.

File: tests/openqa/install/openqa_bootstrap.pm [hunk 2]
```diff
@@ -20,8 +22,65 @@ sub run {
...
+        }
+        else {
+            my $configure = <<'EOC';
+            dbname="${dbname:="openqa"}"
+            dbuser="${dbuser:="geekotest"}"
+            zypper -n install --no-recommends openQA-single-instance qemu-x86 qemu-tools sudo iputils os-autoinst-distri-opensuse-deps
+            [ "$(uname -m)" = "aarch64" ] && zypper -n install --no-recommends qemu-uefi-aarch64 ||:
+            zypper -n search -x qemu-hw-display-virtio-gpu-pci && zypper -n install --no-recommends qemu-hw-display-virtio-gpu qemu-hw-display-virtio-gpu-pci ||:
+            systemctl enable --now postgresql
+            su postgres -c "/usr/share/openqa/script/setup-db" $dbuser $dbname
```
[the remark this answers is not in the document]
Comment (baierjan) [reply tests/openqa/install/openqa_bootstrap.pm:53]: This is copy&paste from https://github.com/os-autoinst/openQA/blob/master/script/openqa-bootstrap#L48

**Comment by okurz** [comment]
@baierjan I saw you applied minor changes according to my in-line comments but have you also seen https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/17216#pullrequestreview-1462700884 ? As this is the main concern

**Comment by baierjan** [comment]
> Except for
> 
> ```
>         add_suseconnect_product('PackageHub') if is_sle;
> ```
> 
> it looks like submitting a new version of openQA including openqa-bootstrap to PackageHub should suffice to not need the custom setup, isn't it?

The upstream version currently fails on some weird dependency stuff (missing `qemu-arm` and `qemu-ppc` on x86_64). But probably yes, it seems that having sufficiently new (and patched) version of openQA-bootstrap should yield the correct behavior and the testing code then could be almost identical to what we currently have in https://github.com/os-autoinst/os-autoinst-distri-openQA/pull/116

**Comment by okurz** [comment]
Ok, so that is the reason why in the ticket I suggested to wait for https://progress.opensuse.org/issues/127541 :)

**Pull request closed without merging**